modules/caddyhttp/rewrite/caddyfile.go

```diff
@@ -25,20 +27,116 @@ func init() {
 
 // parseCaddyfile sets up the handler from Caddyfile tokens. Syntax:
 //
-//     rewrite [<matcher>] <to>
+//     rewrite [<matcher>] [<to>] {
+//         to                <string>
+//         method            <string>
+//         strip_path_prefix <string>
+//         strip_path_suffix <string>
+//         rehandle          <bool>
+//         http_redirect     <status_code>
+//     }
 //
-// The <to> parameter becomes the new URI.
 func parseCaddyfile(h httpcaddyfile.Helper) (caddyhttp.MiddlewareHandler, error) {
-	var rewr Rewrite
+	var (
+		rewr Rewrite
+
+		toIsSet        bool
+		nextBlockIsSet bool
+		rehandleIsSet  bool
+		redirectIsSet  bool
+	)
+
 	for h.Next() {
-		if !h.NextArg() {
+		args := h.RemainingArgs()
+		switch len(args) {
+		case 0:
+		case 1:
+			toIsSet = true
+			rewr.URI = h.Val()
+		default:
 			return nil, h.ArgErr()
 		}
-		rewr.URI = h.Val()
-		if h.NextArg() {
-			return nil, h.ArgErr()
+
+		for h.NextBlock(0) {
+			nextBlockIsSet = true
+			switch h.Val() {
+			case "to":
+				if toIsSet {
+					return nil, h.Err("to is already set")
+				}
+				if !h.Args(&rewr.URI) {
+					return nil, h.ArgErr()
+				}
+			case "method":
+				if !h.Args(&rewr.Method) {
+					return nil, h.ArgErr()
+				}
+				switch rewr.Method {
+				case "GET", "HEAD", "POST", "PUT", "DELETE",
+					"CONNECT", "OPTIONS", "TRACE", "PATCH":
+				default:
+					return nil, h.Errf("unknown method '%s'", rewr.Method)
+				}
+			case "strip_path_prefix":
+				if !h.Args(&rewr.StripPathPrefix) {
+					return nil, h.ArgErr()
+				}
+			case "strip_path_suffix":
+				if !h.Args(&rewr.StripPathSuffix) {
+					return nil, h.ArgErr()
+				}
+			case "rehandle":
+				rehandleIsSet = true
+				if redirectIsSet {
+					return nil, h.Err("http_redirect is already set")
+				}
+
+				var rehandle string
+				if !h.Args(&rehandle) {
+					return nil, h.ArgErr()
+				}
+
+				switch rehandle {
+				case "true":
+					rewr.Rehandle = true
+				case "false":
+				default:
+					return nil, h.Errf("invalid rehandle argument '%s', expected bool", rehandle)
+				}
+			case "http_redirect":
+				redirectIsSet = true
+				if rehandleIsSet {
+					return nil, h.Err("rehandle is already set")
+				}
+
+				var status string
+				if !h.Args(&status) {
+					return nil, h.ArgErr()
+				}
+
+				if len(status) != 3 {
+					return nil, h.Errf("bad status value '%s'", status)
+				}
+
+				statusNum, err := strconv.Atoi(status)
+				if err != nil {
+					return nil, h.Errf("bad status value '%s': %v", status, err)
+				}
+
+				if statusNum < 300 || statusNum > 399 {
+					return nil, h.Errf("bad status value '%s', must be in 3xx range", status)
+				}
```

Comment:
(If we keep this feature)

Is the status code ever assigned to `rewr`? (Also, status code validation will happen shortly after this, as the underlying module is loaded and used, so we don't necessarily have to validate it again here in the Caddyfile.)

Reply:
`rewr.HTTPRedirect` definitely never got set after all that checking.